pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -881,11 +881,15 @@ func podMatchesNodeSelectorAndAffinityTerms(pod *v1.Pod, node *v1.Node) bool {
 
 		// Match node selector for requiredDuringSchedulingIgnoredDuringExecution.
 		if nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
-			nodeSelectorTerms := nodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
-			klog.V(10).Infof("Match for RequiredDuringSchedulingIgnoredDuringExecution node selector terms %+v", nodeSelectorTerms)
-			nodeAffinityMatches = nodeAffinityMatches && nodeMatchesNodeSelectorTerms(node, nodeSelectorTerms)
+			// If pod.Status.Phase is running, which means that this pod was previously scheduled and started running,
+			// the reason running this check is that kubelet is restarted and reruns canAdmitPod method,
+			// in this case, we will ignore NodeAffinity check to follow the *IgnoredDuringExectution* semantic.
+			if pod.Status.Phase != v1.PodRunning {
```

Comment:
How could a pod which is being scheduled be in "running" state? Could you please describe the bug that you are trying to fix?

Reply:
@bsalamat Kubelet also use scheduler's predicates for admitting pods.

Reply:
@bsalamat  sorry for the confusing,  I've updated the description of this PR, You can also take a look at the corresponding issue.